cli/internal/terraform/variables.go

```diff
@@ -19,6 +19,17 @@ type Variables interface {
 	fmt.Stringer
 }
 
+// ClusterVariables should be used in places where a cluster is created.
+type ClusterVariables interface {
+	Variables
+	// TODO (derpsteb): Rename this function once we have introduced an interface for config.Config.
+	// GetCreateMAA does not follow Go's naming convention because we need to keep the CreateMAA property public for now.
+	// There are functions creating Variables objects outside of this package.
+	// These functions can only be moved into this package once we have introduced an interface for config.Config,
+	// since we do not want to introduce a dependency on config.Config in this package.
+	GetCreateMAA() *bool
```

Comment:
Why make this function return a pointer?
As I understand it, callers will only use this function to make a decision based on whether an MAA should be created or not. The output of `GetCreateMAA` should never be parsed into any Terraform files or similar.

Since we can already check if the variable is nil in `AzureClusterVariables.GetCreateMAA`, this would make using the function easier for the caller.